compiler/src/parsing/parser.messages

```diff
@@ -0,0 +1,7448 @@
+
+program: MATCH LPAREN WASMI64 RPAREN LBRACE WASMI64 THICKARROW CONTINUE WHILE
+##
+## Ends in an error in state: 745.
+##
+## lseparated_nonempty_list_inner(comma,match_branch) -> lseparated_nonempty_list_inner(comma,match_branch) . comma match_branch [ RBRACE EOL COMMA ]
+## match_branches -> lseparated_nonempty_list_inner(comma,match_branch) . option(comma) [ RBRACE EOL ]
+##
+## The known suffix of the stack is as follows:
+## lseparated_nonempty_list_inner(comma,match_branch)
+##
+
+Expected a comma followed by another match branch (or a '}' to complete the match expression).
+
+program: MATCH LPAREN WASMI64 RPAREN LBRACE WASMI64 THICKARROW WASMI64 COMMA EOL WHILE
+##
+## Ends in an error in state: 747.
+##
+## lseparated_nonempty_list_inner(comma,match_branch) -> lseparated_nonempty_list_inner(comma,match_branch) comma . match_branch [ RBRACE EOL COMMA ]
+## option(comma) -> comma . [ RBRACE EOL ]
+##
+## The known suffix of the stack is as follows:
+## lseparated_nonempty_list_inner(comma,match_branch) comma
+##
+## WARNING: This example involves spurious reductions.
+## This implies that, although the LR(1) items shown above provide an
+## accurate view of the past (what has been recognized so far), they
+## may provide an INCOMPLETE view of the future (what was expected next).
+## In state 3, spurious reduction of production nonempty_list(eol) -> EOL
+## In state 5, spurious reduction of production eols -> nonempty_list(eol)
+## In state 89, spurious reduction of production comma -> COMMA eols
+##
+
+Expected another match branch (or a '}' to complete the match expression).
+
+program: MATCH LPAREN WASMI64 RPAREN LBRACE WASMI64 WHEN VOID EOF
+##
+## Ends in an error in state: 739.
+##
+## match_branch -> pattern option(when_guard) . thickarrow expr [ RBRACE EOL COMMA ]
+##
+## The known suffix of the stack is as follows:
+## pattern option(when_guard)
+##
+## WARNING: This example involves spurious reductions.
+## This implies that, although the LR(1) items shown above provide an
+## accurate view of the past (what has been recognized so far), they
+## may provide an INCOMPLETE view of the future (what was expected next).
+## In state 41, spurious reduction of production non_assign_expr -> simple_expr
+## In state 120, spurious reduction of production non_binop_expr -> non_assign_expr
+## In state 64, spurious reduction of production annotated_expr -> non_binop_expr
+## In state 238, spurious reduction of production non_stmt_expr -> annotated_expr
+## In state 60, spurious reduction of production expr -> non_stmt_expr
+## In state 737, spurious reduction of production when_guard -> WHEN expr
+## In state 738, spurious reduction of production option(when_guard) -> when_guard
+##
+
+Expected an arrow (=>) followed by an expression.
+
+program: MATCH LPAREN WASMI64 RPAREN LBRACE WASMI64 WHILE
+##
+## Ends in an error in state: 735.
+##
+## match_branch -> patte
```

Comment:
See above about "type expression", though this one bothers me less for some reason